dev/storage_base.h

```diff
@@ -124,8 +124,27 @@ namespace sqlite_orm {
                 return sqlite3_busy_timeout(con.get(), ms);
             }
 
+            /*
+             * Ownership is NOT transferred, i.e. data
+             * must not go out of scope for the lifetime of the database.
+             */
+            template<typename T = void>
+            int busy_handler(std::function<int(T *, int)> f, T &data) {
```

Comment:
I don't understand why do you need this function to be template. Actually `std::function` allows us binding any data we want so it is enough. Dedicated argument for custom data is not required - it can be stored inside lambdas capture. `std::function` must be a member inside `storage_base`. Next in `on_open_internal` you must call `sqlite3_busy_handler` if this member is not null. Custom data argument must be a pointer to a storage and callback that is must be passed to `sqlite3_busy_handler` function must be a static private/protected function in `storage_base` class. Why this idea is better? Cause it has less templates. Templates must exist only if we cannot avoid them. The second is custom data is stored inside `std::function` and this is ok - it solves the issue from the first comment. I'd like you to refactored your solution as I described here. Of course if you have a different opinion or a better solution please feel free to comment it here or email me (the first option is more preferred cause conversation will be available for everyone). If you want I can implement it on my own today or tomorrow.

Reply:
I have chosen to write it this way for one simple reason: Because it directly correlates to the function signature documented as sqlite, `int(void*, int)`. 
The user of this interface may check `sqlite3_busy_handler`, see that it asks for a function `int(void*, int)`, and see that this interface does not expose that functionality. It is not obvious that the idea is to capture data in the lambdas.
I have seen your merge request. It is good, but please add documentation about the intention.